src/pipeline.ts

```diff
@@ -320,9 +331,17 @@ export class GitHubWorkflow extends PipelineBase {
       });
     };
 
+    const replaceAssetHash = (template: string) => {
+      const hash = path.parse(template.split('/').pop() ?? '').name;
+      if (this.assetHashMap[hash] === undefined) {
+        throw new Error(`Template asset hash ${hash} not found.`);
+      }
+      return template.replace(hash, `\${{ needs.${this.assetHashMap[hash]}.outputs.asset-hash }}`);
+    };
+
     const params: Record<string, any> = {
       'name': stack.stackName,
-      'template': resolve(stack.templateUrl),
+      'template': replaceAssetHash(resolve(stack.templateUrl)),
```

Comment:
I think what I imagined (and I think we discussed) this code could look like, is something like:

```ts
const outputName = sanitizeName(`a_${stack.artifactId}`);
console.log(`::set-output name=${ouputName}::${stack.templateUrl}`);
const params: Record<string, any> = {
  'template': `\${{ synth.outputs.${outputName} }}`,
};
```

What do you think about this?

Reply:
talked about how this would fail the use case where we upload `cdk.out` directly to github and skip `cdk synth`.